upapi/user/__init__.py

```diff
@@ -29,8 +30,18 @@ def __init__(self, *args, **kwargs):
     @property
     def friends(self):
         """
+        If not cached yet, cache the Friends object. Return the Friends object.
+
+        :return: a Friends object
+        """
+        if self._friends is None:
+            self._friends = self.get_friends()
+        return self._friends
```

Comment:
I was thinking you would either just make a new object, or call get_friends directly, but I think I'll make a change where get_friends also sets self._friends